### PR TITLE
[DOC] Enhanced RDoc for Time

### PR DESCRIPTION
**Pull request by BurdetteLamar**
Revises the introductory doc for Time, in particular adding a small section "Epoch Seconds" that is now linked to from other parts of the doc.

Adds "Related" remarks to several methods' doc.

### CODE REVIEW

File: time.c
Comment (jeremyevans) [inline time.c:3501]: I would remove `Time.at`.  Since `to_f` and `to_r` are instance methods and not class methods, you should use `#to_f` and `#to_r` instead of `Time.to_f` and `Time.to_r`.  I don't think there is significant value in describing the difference between `to_f` and `to_r` here.  I would use: `Related: #to_f, #to_r`.  Similar change for `to_f` and `to_r` method documentation.

File: timev.rb
Comment (jeremyevans) [inline timev.rb:31]: I would guess that is not always true, for  `time_to_r` in `time.c` has this code:

```c
    GetTimeval(time, tobj);
    v = rb_time_unmagnify_to_rational(tobj->timew);
    if (!RB_TYPE_P(v, T_RATIONAL)) {
        v = rb_Rational1(v);
    }    
    return v;
```

So I'm guessing there is at least a possibility that that is not stored as a rational always.  My guess would be that it is stored as an integer if there are no subseconds.

In any case, I don't think it makes sense to discuss implementation details such as how values are stored internally.  You can mention that `to_r` should always provide an exact time, whereas `to_i` and `to_f` may not for some times.  Other documentation changes that mentions how values are stored internally should also be updated.

File: timev.rb
Comment (jeremyevans) [inline timev.rb:41]: `#to_i` and `#to_f`.  `ClassName.method` is for class/singleton methods, `ClassName#method` is for instance methods.

File: timev.rb
Comment (jeremyevans) [inline timev.rb:50]: This existed before, and was important when Ruby transitioned from using GMT to UTC (in Ruby 1.8.0, released in 2003), but I'm not actually sure how much value it has now.  I would vote to drop it.

**Review by jeremyevans — CHANGES_REQUESTED** [review]
Thanks for working on this, please see inline changes.

File: timev.rb
Comment (BurdetteLamar) [reply timev.rb:323]: All deal with Epoch seconds, so ::at is sort of an inverse to the others.

Comment (BurdetteLamar) [reply timev.rb]: Removed.

File: time.c
Comment (BurdetteLamar) [reply time.c]: Fixed.

File: time.c
Comment (BurdetteLamar) [reply time.c]: Fixed.

File: timev.rb
Comment (BurdetteLamar) [reply timev.rb]: Done (two places).

File: timev.rb
Comment (BurdetteLamar) [reply timev.rb]: Fixed.

File: timev.rb
Comment (BurdetteLamar) [reply timev.rb]: Removed.

**Review by jeremyevans — APPROVED** [review]

File: timev.rb
Comment (jeremyevans) [inline timev.rb:31]: This is missing `#` for comments.

**Review by jeremyevans — CHANGES_REQUESTED** [review]
Sorry, missed that this introduced non-documentation changes that broke loading the `timev.rb` file

**Comment by k0kubun** [comment]
@BurdetteLamar The CI is failing on master because of this. Can you address that or revert this for now?

**Comment by k0kubun** [comment]
Thanks Jeremy. https://github.com/ruby/ruby/commit/055fc7b14dc375089b454cc7eaebeb78f9f3efaa

**Comment by jeremyevans** [comment]
I fixed this.  @BurdetteLamar please make sure CI is passing before merge (at least mostly passing, don't have to wait for all jobs to finish), even if you have approval for the documentation changes.

**Comment by BurdetteLamar** [comment]
@k0kubun and @jeremyevans, thanks.  I'll be more alert.

**Pull request merged**
